app/scripts/controllers/user-storage/encryption/encryption.ts

```diff
@@ -127,6 +150,35 @@ class EncryptorDecryptor {
     // Decrypt and return result.
     return gcm(key, nonce).decrypt(ciphertext);
   }
+
+  #getOrGenerateScryptKey(
+    password: string,
+    o: EncryptedPayload['o'],
+    salt?: Uint8Array,
+  ) {
+    const cachedKey = salt ? getCachedKeyBySalt(salt) : getAnyCachedKey();
```

Comment:
@Prithpal-Sooriya, why is `getAnyCachedKey()` needed?

Reply:
It's needed because the encryption only needs one key, whichever it is; in normal circumstances, there's only one - most recently created. Older payloads or ones coming from other instances are encrypted with different locally derived keys with a different salt, so they need to be derived here as well for decryption to succeed. In case the same key/salt is used for another thing to decrypt from the other instance, it's also cached. For encrypting it doesn't really matter much which of the keys is used, so the first one we can get from the cache is fine.

Reply:
This is a good callout.
Currently in all cases we are using the same password.

I think we can change this to be a map to password + salt to receive the cached key :)